lib/src/sidenav/Sidenav.tsx

```diff
@@ -67,49 +156,156 @@ const SidenavContainer = styled.div<SidenavPropsType>`
-  font-family: ${(props) => props.theme.subtitleFontFamily};
-  font-size: ${(props) => props.theme.subtitleFontSize};
-  font-style: ${(props) => props.theme.subtitleFontStyle};
-  font-weight: ${(props) => props.theme.subtitleFontWeight};
-  color: ${(props) => props.theme.subtitleFontColor};
-  letter-spacing: ${(props) => props.theme.subtitleFontLetterSpacing};
-  text-transform: ${(props) => props.theme.subtitleFontTextTransform};
-  margin-bottom: 4px;
+const SidenavGroup = styled.div`
+  width: 100%;
+  a {
+    padding: 7px 24px 7px 36px;
+  }
 `;
 
-const SidenavMenuLink = styled.a`
-  font-family: ${(props) => props.theme.linkFontFamily};
-  font-size: ${(props) => props.theme.linkFontSize};
-  font-style: ${(props) => props.theme.linkFontStyle};
-  font-weight: ${(props) => props.theme.linkFontWeight};
-  color: ${(props) => props.theme.linkFontColor};
+const SidenavGroupTitle = styled.span`
+  svg {
+    width: 16px;
+    margin-right: 8px;
+  }
+  box-sizing: border-box;
+  width: 100%;
+
+  font-family: ${(props) => props.theme.groupTitleFontFamily};
+  font-style: ${(props) => props.theme.groupTitleFontStyle};
+  font-weight: ${(props) => props.theme.groupTitleFontWeight};
+  font-size: ${(props) => props.theme.groupTitleFontSize};
+  line-height: 19px;
+
+  display: flex;
+  align-items: center;
+  margin: 0px;
+  padding: 7px 24px;
+`;
+
+const SidenavGroupTitleButton = styled.button`
+  all: unset;
+  cursor: pointer;
+  justify-content: space-between;
+  box-sizing: border-box;
+  width: 100%;
+
+  font-family: ${(props) => props.theme.groupTitleFontFamily};
+  font-style: ${(props) => props.theme.groupTitleFontStyle};
+  font-weight: ${(props) => props.theme.groupTitleFontWeight};
+  font-size: ${(props) => props.theme.groupTitleFontSize};
+  line-height: 19px;
+
+  display: flex;
+  align-items: center;
+  margin: 0px;
+  padding: 7px 24px;
+  &:focus {
+    outline: 2px solid ${(props) => props.theme.linkFocusColor};
+    outline-offset: -2px;
+  }
+  &:hover {
+    background-color: ${(props) => props.theme.groupTitleHoverBackgroundColor};
+  }
+  &:active {
+    background-color: ${(props) => props.theme.groupTitleActiveBackgroundColor};
+  }
+
+  svg {
+    width: 18px;
+    height: auto;
+  }
+`;
+
+type StyledLinkProps = {
+  selected: boolean;
+};
+const SidenavLink = styled.a<StyledLinkProps>`
   letter-spacing: ${(props) => props.theme.linkFontLetterSpacing};
+
   text-transform: ${(props) => props.theme.linkFontTextTransform};
   text-decoration: ${(props) => props.theme.linkTextDecoration};
-  margin: ${(props) =>
-    `${props.theme.linkMarginTop} ${props.theme.linkMarginRight} ${props.theme.linkMarginBottom} ${props.theme.linkMarginLeft}`};
+
+  font-family: ${(props) => props.theme.linkFontFamily};
+  font-style: ${(props) => props.theme.linkFontStyle};
+  font-weight: ${(props) => props.theme.linkFontWeight};
+  font-size: ${(props) => props.theme.linkFontSize};
+  line-height: 19px;
+
```

Comment:
Same here. Why this value for the line-height? We never use odd values, like 19px.